rmgpy/tools/uncertainty.py

```diff
@@ -341,37 +326,65 @@ def loadModel(self, chemkinPath, dictionaryPath, transportPath=None):
                                                               dictionaryPath=dictionaryPath,
                                                               transportPath=transportPath)
 
+    def retrieveSaturatedSpeciesFromList(self,species):
+        """
+        Given a radical `species`, this function retrieves the saturated species objects from a list of species objects.
+        """
+        
+        molecule = species.molecule[0]
+        assert molecule.isRadical(), "Method only valid for radicals."
+        saturatedStruct = molecule.copy(deep=True)
+        saturatedStruct.saturate()
+        for otherSpecies in self.speciesList:
+            if otherSpecies.isIsomorphic(saturatedStruct):
+                return otherSpecies, False
+        else:
+            for key,value in self.database.thermo.libraries.iteritems():
```

Comment:
It might be better to create the species first, then call `self.database.thermo.getThermoDataFromLibraries()`.

As an additional note, the species can be directly using `newSpc = Species(molecule=[saturatedStruct])` instead of conversion via adjlist.

Reply:
This should be much cleaner now